internal/storage/distlock_test.go

```diff
@@ -184,3 +165,233 @@ func TestDistributedLock(t *testing.T) {
 			distLockProvider.GetDuration().String())
 	}
 }
+
+func TestDistrubutedLockAlreadyAcquired(t *testing.T) {
+	lockDur := 5 * time.Second
+	err := distLockProvider.DistributedTimedLock(lockDur)
+	if err != nil {
+		t.Errorf("DistributedTimedLock() failed: %v", err)
+	}
+
+	// Attempt to acquire the lock again
+	err = distLockProvider.DistributedTimedLock(1 * time.Second)
+	if err == nil {
+		t.Error("Expected error when trying to acquire lock again, but got none")
+	}
+}
+
+func TestDistributedLockTimeout(t *testing.T) {
+	lockDur := 4 * time.Second
+
+	err := distLockProvider.DistributedTimedLock(lockDur)
+	if err != nil {
+		t.Errorf("DistributedTimedLock() failed: %v", err)
+	}
+
+	// Attempt to acquire the lock again with a different provider
+	distLockProvider2, err := createDistLockProvider()
+	if err != nil {
+		t.Errorf("Error creating second distributed lock provider: %v", err)
+	}
+	// Initialize the second distributed lock provider
+	err = distLockProvider2.Init(logrus.New())
+	if err != nil {
+		t.Errorf("Error initializing second distributed lock provider: %v", err)
+	}
```

Comment:
https://pkg.go.dev/github.com/stretchr/testify@v1.10.0/require#NoErrorf will cut down a bit on Go's `if err != nil { fail }` verbosity in tests, since they tend to do many error checks in a row. That's optional though, the existing `if` blocks are fine if you don't want to bother.